### DISCUSSION TITLE
Open a pdf file to a bookmark destination with command line

### DISCUSSION THREAD

**Discussion by Valter62** [sumatrapdfreader/sumatrapdf#4592] [General]
I have a pdf file  MA-MO-AR_I.pdf
In the file there is the bookmark    CMD_028_1
Sumatra is closed
I use the command line :
SumatraPDF.exe -reuse-instance -named-dest CMD_028_1 MA-MO-AR_I.pdf
Sumatra open the file but it doesn't scroll it to the bookmark assigned.
how can i do?
Thanks

**Comment by GitHubRulesOK** [comment]
provide the file or a link to the file as the only way to see why it may not be a nameddest is to  look inside the file at that structure

![image](https://github.com/user-attachments/assets/87c6c355-c121-494a-a451-8686214b2d60)
![image](https://github.com/user-attachments/assets/728d4cfa-15f1-43b4-a38b-911ee0353fed)

**Reply by Valter62** [reply]
[MA-MO-AR_I.pdf](https://github.com/user-attachments/files/17399054/MA-MO-AR_I.pdf)

Thanks

**Comment by GitHubRulesOK** [comment]
Hmm works for me with old XP version 
![image](https://github.com/user-attachments/assets/87b2f535-6667-4484-bdf0-08569faf5364)

But only top of page 17  for 3.6 pre-release and that has been raised before as a recent problem
![image](https://github.com/user-attachments/assets/1d141fd9-e4fc-4fa9-bc93-39b1f94208f5)


Supposedly was corrected last year in mupdf but has been a does / does not work intermittent problem for some time
https://github.com/sumatrapdfreader/sumatrapdf/issues/3426

**Comment by Xerxes-77** [comment]
I have a similar question. 
Running `sumatrapdf -named-dest "1.8" manual.pdf` works the first time and correctly goes to bookmark 1.8. However if I scroll around the PDF, close Sumatra and then run the same command, the PDF opens in the last viewed page instead of bookmark 1.8.

I'm a newbie to Sumatra but apparently this is related to a setting that remembers the last session and this overrides the scroll to the -named-dest when invoked.

If instead of closing Sumatra I close the document tab, then it will reopen correctly at the -named-dest position when invoked again from the command line.

**Comment by GitHubRulesOK** [comment]
@Xerxes-77 

yes the history will take you back to last viewed page  same as the similar settings in Edge or Acrobat.

With out me checking I am unsure if it simply the one in options or if its another, so simplest for you to try is uncheck this one
![image](https://github.com/user-attachments/assets/eef87fd4-0827-4780-98dc-1b76f154b260)

**Reply by Xerxes-77** [reply]
Thank you, the thing is that I'm invoking SumatraPDF from an application so I don't have direct access to the options... I need to open a fresh instance of Sumatra and go directly to a named destination.

SumatraPDF v3.4.6 works fine because it doesn't automatically save the session data in the settings.txt file, however v3.5.2 does. The main problem is that this overrides the -named-dest parameter.

**Reply by GitHubRulesOK** [reply]
Most applications as programmable should be able to edit the settings.txt (find and replace) but you may be able to do that other ways.  

I have not tried these out but via DDE command you can try, SEE https://www.sumatrapdfreader.org/docs/DDE-Commands
- CmdClearHistory
- CmdForgetSelectedDocument
- CmdOptions

**`SumatraPDF.exe -dde "[Cmd...... (\"blah\",\"blah\",##)]"`**
